pkg/featuregate/featuregate.go

```diff
@@ -0,0 +1,73 @@
+package featuregate
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+)
+
+// FeatureGate provides methods to check state of the feature
+type FeatureGate interface {
+	// IsEnabled returns state of the feature,
+	// if feature name is unknown will always return false
+	IsEnabled(feature string) bool
+	// Init set state for the features from the provided map.
+	// completely removes the previous state
+	Init(features map[string]bool)
+	// Set state of the specific feature, keeps existing state for other features
+	Set(feature string, state bool)
```

Comment:
Same here, `Set` seems to be used in tests only. How is it going to be used?

Reply:
@zeeke I removed `Set` function